R/data_rename.R

```diff
@@ -96,28 +97,34 @@
 #'
 #' @export
 data_rename <- function(data,
-                        pattern = NULL,
+                        select = NULL,
                         replacement = NULL,
                         safe = TRUE,
                         verbose = TRUE,
+                        pattern = NULL,
                         ...) {
+  if (!is.null(pattern)) {
+    .is_deprecated("pattern", "select")
+    select <- pattern
+  }
+
   # change all names if no pattern specified
-  if (is.null(pattern)) {
-    pattern <- names(data)
+  if (is.null(select)) {
+    select <- names(data)
   }
 
-  if (!is.character(pattern)) {
-    insight::format_error("Argument `pattern` must be of type character.")
+  if (!is.character(select)) {
+    insight::format_error("Argument `select` must be of type character.")
   }
```

Comment:
This could be removed, as `pattern` (or now: `select`) no longer needs to be a name as string.